Assets/Plugins/CountlySDK/Services/ConsentCountlyService.cs

```diff
@@ -1,47 +1,310 @@
+                }
+                return;
+            }
+
+            foreach (string name in groupName) {
+                Features[] features = _countlyFeatureGroups[name];
+                foreach (Features feature in features) {
+                    GiveConsentInternal(feature);
+                }
+            }
+
+            NotifyListeners();
+        }
+
+        /// <summary>
+        /// Remove consent of a group of features
+        /// </summary>
+        /// <param name="groupName">name of the consent group</param>
+        /// <returns></returns>
+        public void RemoveConsentOfFeatureGroup(string[] groupName)
+        {
+            if (!RequiresConsent) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Enable Consents");
+                }
+
+                return;
+            }
+
+            if (groupName == null) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Calling RemoveConsentOfFeatureGroup with null groupName!");
+                }
+                return;
+            }
+
+            foreach (string name in groupName) {
+                Features[] features = _countlyFeatureGroups[name];
+                foreach (Features feature in features) {
+                    RemoveConsentInternal(feature);
+                }
+            }
+
+            NotifyListeners();
+        }
+        #endregion
+
+        #region Helper Methods
+
+        private void GiveConsentInternal(Features feature)
+        {
+            if (_countlyFeatureConsents.ContainsKey(feature)) {
+                _countlyFeatureConsents[feature] = true;
+
+            } else {
+                _countlyFeatureConsents.Add(feature, true);
+            }
+
+            if (_config.EnableConsoleLogging) {
+                Debug.Log("[Countly] Setting consent for feature: [" + feature.ToString() + "] with value: [true]");
+            }
+        }
+
+        private void GiveConsentInternal(Features[] features)
+        {
+            if (!RequiresConsent) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Enable Consents");
+                }
+
+                return;
+            }
+
+            if (features == null) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Calling GiveConsent with null features list!");
+                }
+
+                return;
+            }
+            //Remove Duplicates entries
+            features = features.Distinct().ToArray();
+
+            foreach (Features feature in features) {
+                GiveConsentInternal(feature);
+            }
+        }
+
+        private void RemoveConsentInternal(Features feature)
```

Comment:
This function is basicly a duplicate of the "GiveConsentInternal" function. Each just uses a slightly different bool value.
Combine them both into one function where a bool value for the new consent value is provided.